plugins/woocommerce-blocks/packages/interactivity-components/dropdown/index.ts

```diff
@@ -19,74 +24,74 @@ export type DropdownContext = {
 	isOpen: boolean;
 };
 
-store( 'woocommerce/interactivity-dropdown', {
+type DropdownStore = {
 	state: {
-		get placeholderText() {
-			const context = getContext< DropdownContext >();
-			const { selectedItem } = context;
-
-			return selectedItem.label || 'Select an option';
-		},
-		get isSelected() {
-			const context = getContext< DropdownContext >();
-
-			const {
-				currentItem: { value },
-			} = context;
+		selectedItem?: {
+			label: string | null;
+			value: string | null;
+		};
+		placeholderText: string;
+		isSelected: boolean;
+	};
```

Comment:
This may be out of scope of this PR but the state are synced between all dropdown components on the same page:


https://github.com/woocommerce/woocommerce/assets/5423135/f61c63c5-ffb8-4f74-b3ac-589f3ad71c11

Reply:
Oh dear. That's quite concerning. I should probably try fix that here

Reply:
@dinhtungdu on second thought, would you mind if I do this in follow up. My reasoning is I'm currently working on the multi select and so I can fix this issue in that PR.

Reply:
>  would you mind if I do this in follow up. My reasoning is I'm currently working on the multi select and so I can fix this issue in that PR.

No issue at all :D